nginx-shibboleth/config.sls

```diff
@@ -1,15 +1,49 @@
 {% from "nginx-shibboleth/map.jinja" import nginx_shibboleth with context %}
+{% set pillar_config = salt.pillar.get('nginx-shibboleth:config', {}) %}
 
 include:
   - .install
   - .service
 
-nginx-shibboleth-config:
+{% if nginx_shibboleth.generate_sp_cert %}
+generate_shibboleth_sp_cert_and_key:
+  cmd.run:
+    - name: shib-keygen
+    - creates: /etc/shibboleth/sp-cert.pem
+{% else %}
+{% for secret in ['key', 'cert'] %}
+place_shibboleth_sp_{{ secret }}:
   file.managed:
-    - name: {{ nginx_shibboleth.conf_file }}
-    - source: salt://nginx-shibboleth/templates/conf.jinja
-    - template: jinja
-    - watch_in:
-      - service: nginx-shibboleth_service_running
+    - name: /etc/shibboleth/sp-{{ secret }}.pem
+    - mode: 0600
+    - user: _shibd
+    - group: _shibd
+    - contents_pillar: nginx-shibboleth:secrets:{{ secret }}
```

Comment:
Not exactly sure how this will be rendered as I don't see `secrets` defined in `map.jinja` or the pillar overrides

Reply:
If the values are present in the pillar at nginx-shibboleth:secrets:key/cert then the contents of those values will be rendered out. Otherwise it will generate a set of certificate and key.